crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -195,13 +195,89 @@ impl SettlementEncoder {
                 )?
             }
             OrderClass::Limit => {
```

Comment:
This comes from the fact that the `executed_amount` is kind of meaningless for this order type because solvers can't know the actual executed amount. After all we gave them a synthetic order with adjusted `sell_amount`.